### PR TITLE
HDDS-5710. initialize sequenceIdToLastIdMap when SequenceIdGenerator#StateManager reinitializes

### PR DESCRIPTION
**Pull request by JacksonYao287**
## What changes were proposed in this pull request?

2021-09-03 12:50:38,632 [1114efaf-c975-4ea0-bc8b-c3309a57cc8d@group-58D600BFB107-StateMachineUpdater] WARN org.apache.hadoop.hdds.scm.ha.SequenceIdGenerator: Failed to allocate a batch for localId, expected lastId is 0, actual lastId is 107544261438510000.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-5710

## How was this patch tested?

current unit test

### CODE REVIEW

**Comment by JacksonYao287** [comment]
seems a flaky case , not caused by this patch!

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SequenceIdGenerator.java [hunk 1]
```diff
@@ -253,9 +254,28 @@ public Long getLastId(String sequenceIdName) {
...
+    private void initialize() throws IOException {
+      TableIterator<String,
+          ? extends Table.KeyValue<String, Long>>
+          iterator = sequenceIdTable.iterator();
+
+      while (iterator.hasNext()) {
+        Table.KeyValue<String, Long> kv = iterator.next();
+        final String sequenceIdName = kv.getKey();
+        final Long lastId = kv.getValue();
+        org.apache.ratis.util.Preconditions.assertNotNull(sequenceIdName,
```
Comment (bshashikant) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SequenceIdGenerator.java:273]: Can u fix the import and use just Preconditions check?

**Comment by bshashikant** [comment]
Can you explain the problem in bit detail here?

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SequenceIdGenerator.java (hunk 1 above)
Comment (JacksonYao287) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SequenceIdGenerator.java:273]: sure, will do this

**Review by bshashikant — APPROVED** [review]
Thanks @JacksonYao287 for the explanatio and the contribution.

**Comment by JacksonYao287** [comment]
@bshashikant  thanks for the review

**Pull request merged**